vermouth/utils.py

```diff
@@ -127,3 +141,53 @@ def are_all_equal(iterable):
     iterator = iter(iterable)
     first = next(iterator, None)
     return all(item == first for item in iterator)
+
+
+def are_different(left, right):
+    """
+    Return True if two values are different from one another.
+
+    Values are considered different if they do not share the same type. In case
+    of numerical value, the comparison is done with :func:`numpy.isclose` to
+    account for rounding. In the context of this test, `nan` compares equal to
+    itself, which is not the default behavior.
+    """
+    if left.__class__ != right.__class__:
+        return True
+    
+    if left is None:
+        return False
+
+    if isinstance(left, numbers.Number):
+        try:
+            return not np.isclose(left, right, equal_nan=True)
+        except TypeError:
+            # Some things pretend to be numbers but cannot go through isclose.
+            # It is the case of integers that overflow an int64 for instance.
+            return left != right
+    
+    if isinstance(left,(str, bytes)):
+        return left != right
+
+    filler = _Filler()
+
+    if isinstance(left, collections.abc.Mapping):
+        zipped = itertools.zip_longest(left.items(), right.items(), fillvalue=filler)
+        return any(
+            item_left is filler
+            or item_right is filler
+            or are_different(item_left[0], item_right[0])  # keys
+            or are_different(item_left[1], item_right[1])  # values
+            for item_left, item_right in zipped
+        )
+
+    if isinstance(left, collections.abc.Iterable):
+        zipped = itertools.zip_longest(left, right, fillvalue=filler)
+        return any(
+            item_left is filler
+            or item_right is filler
+            or are_different(item_left, item_right)
```

Comment:
You could trim this a bit by making _Filler object always compare False. But it's also fine to leave it like this.

Reply:
I agree that it would work. I like how explicit this looks, though.